Packs/MandiantAdvantageAttackSurfaceManagement/Integrations/AttackSurfaceManagement/AttackSurfaceManagement.py

```diff
@@ -0,0 +1,520 @@
+# TODO: REMOVE the following dummy command function
+def baseintegration_dummy_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    dummy = args.get('dummy', None)
+    if not dummy:
+        raise ValueError('dummy not specified')
+
+    # Call the Client function and get the raw response
+    result = client.baseintegration_dummy(dummy)
+
+    return CommandResults(
+        outputs_prefix='BaseIntegration',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+# TODO: ADD additional command functions that translate XSOAR inputs/outputs to Client
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    params = demisto.params()
+    # get the service API url
+    base_url = urljoin(params['url'], '/api/v1')
+
+    # if your Client class inherits from BaseClient, SSL verification is
+    # handled out of the box by it, just pass ``verify_certificate`` to
+    # the Client constructor
+    verify_certificate = not params.get('insecure', False)
+
+    # if your Client class inherits from BaseClient, system proxy is handled
+    # out of the box by it, just pass ``proxy`` to the Client constructor
+    proxy = params.get('proxy', False)
+
+    access_key = params.get('credentials', {}).get('identifier')
+    secret_key = params.get('credentials', {}).get('password')
+    project_id = params.get('project_id')
+    collections_raw = params.get('collection_ids', '').split(',')
+    collections = [c.strip() for c in collections_raw]
+
+    limit = int(params.get("max_fetch", 50))
+    timeout = int(params.get("timeout", 60))
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+
+        # TODO: Make sure you add the proper headers for authentication
+        # (i.e. "Authorization": {api key})
+        client = Client(
+            access_key=access_key,
+            secret_key=secret_key,
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy,
+            timeout=timeout,
+            limit=limit,
+            project_id=project_id,
+            collection_ids=collections
+        )
+
+        commands_with_args: dict[str, typing.Callable] = {
+            'get-remote-data': get_remote_data_command,
+            'update-remote-system': update_remote_system_command,
+            'attacksurfacemanagement-get-projects': get_projects,
+            'attacksurfacemanagement-get-collections': get_collections
+        }
+        commands_no_args: dict[str, typing.Callable] = {
+            'test-module': test_module
+        }
+
+        command = demisto.command()
+
+        if command in commands_no_args:
+            # This is the call made when pressing the integration Test button.
+            result = commands_no_args[command](client)
+            return_results(result)
+
+        elif command in commands_with_args:
+            args = demisto.args()
```

Comment:
call demisto.args() in main and re-use the variable

Reply:
This is the `main` function - I'm just only calling `args` if the command requires/accepts `args`.  I can shift it up to be with the `proxy` and other configuration though

Reply:
not super important, but there is no harm that `demisto.args()` will be called once in the main even if its not being used